src/Pages/AuthPages/LoginPage.jsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { Link, useNavigate, useLocation } from "react-router-dom";
+import { getUserCall } from "../../ApiCalls";
+import { useAuth } from "../../Context/Index";
+import "./Auth.css";
+
+export const LoginPage = () => {
+  const { setIsUserLoggedIn, setUserInfo, isUserLoggedIn, userInfo } =
+    useAuth();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const loginDetailsHandler = async (event) => {
+    event.preventDefault();
+    let [email, password] = event.target.elements;
+    const { data, status } = await getUserCall(email.value, password.value);
+    if (status === 200) {
+      setIsUserLoggedIn(true);
+      setUserInfo({ ...data.foundUser });
+      localStorage.setItem("userLoginToken", data.encodedToken);
+      navigate(location?.state?.from?.pathname || "/video-listing-page");
+    }
+  };
+
+  const guestLoginDetailsHandler = async (email, password) => {
+    const { data, status } = await getUserCall(email, password);
+    if (status === 200 || status === 201) {
+      setIsUserLoggedIn(true);
+      setUserInfo({ ...data.foundUser });
+      localStorage.setItem("userLoginToken", data.encodedToken);
+      navigate(location?.state?.from?.pathname || "/video-listing-page");
+    }
```

Comment:
This section looks repeated and can be reduced by extracting the status part into a separate function giving it the parameter of data and status

Reply:
cool, I will add this.